packages/pluginmanager-extension/src/index.ts

```
@@ -47,11 +48,12 @@ const pluginmanager: JupyterFrontEndPlugin<IPluginManager> = {
   id: PLUGIN_ID,
   description: 'Enable or disable individual plugins.',
   autoStart: true,
-  requires: [],
+  requires: [JupyterLab.IInfo],
   optional: [ITranslator, ICommandPalette, ILayoutRestorer],
   provides: IPluginManager,
   activate: (
-    app: JupyterLab,
+    app: JupyterFrontEnd,
+    appInfo: JupyterLab.IInfo,
```

I guess the added value is in being able to override the available plugin list?

I do not follow the use case here (not very well).

---

Yes, since most plugins expect a `JupyterFrontEnd` and not a `JupyterLab` as the first argument of the `activate` function.

There is some information about the motivation in the description above:

> Instead we can require JupyterLab.IInfo, which provides the information needed for the plugin. That way another lab-based application could provide JupyterLab.IInfo via a plugin, instead of being itself a JupyterLab.

---

If a lab-based application is not a `JupyterLab`, then why should it provide `JupyterLab.IInfo`? Shouldn't it be like `JupyterFrontEnd.IInfo` token? I guess I am not suggesting a change needs to be made this, I just mean that this is not much of an improvement.

I suspect that you mean something like Notebook 7 should not need to inherit from `JupyterLab` class. Technically it does not need to inherit as long as it has `.info`, but having this as an implicit required attribute is indeed not great. Having a required token is a little better because the plugin will fail a little bit earlier, but it does not solve the "I am not a JupyterLab, I just quack like one" issue, nor does it guarantee correctness downstream as ultimately if downstream is expected to provide `JupyterLab.IInfo` but they are not a `JupyterLab`, they will need to provide some dummy values and might fail when attempting to do so.

If you think we should move forward with this token as-is, I am happy to approve and merge, but also curious if you thought of other solutions.

---

This PR tries to slightly improve the current situation and make the "requirement" on that `IInfo` object a bit less dependent on the `app` being provided to the `activate` function.

Indeed, in the case of Notebook 7 (or other lab-based applications), this information could be provided as a separate plugin instead of having it as a mandatory property on the `app`. Although it's true it can be weird to have to define a `JupyterLab.IInfo` if the app is not a `JupyterLab` app itself.

This makes it sightly easier for downstreams, but agreed that in the case of JupyterLab it does not change the situation that much.